core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/downscale/StackDownscaleService.java

```diff
@@ -113,7 +118,32 @@ public void finishStackDownscale(StackScalingFlowContext context, Collection<Lon
         flowMessageService.fireEventAndLog(stack.getId(), AVAILABLE.name(), STACK_DOWNSCALE_SUCCESS, String.join(",", deletedInstanceIds));
     }
 
-    private void fillDiscoveryFQDNForRepair(StackView stack, List<InstanceMetadataView> removableInstances) {
+    private void putDanglingVolumeSetsToFailedState(StackView stack) {
+        List<Resource> diskResources = resourceService.findByStackIdAndType(stack.getId(), stack.getDiskResourceType());
+        Map<Resource, Optional<VolumeSetAttributes>> volumeSetAttributesOptionalMap = diskResources.stream().collect(Collectors.toMap(volumeSet -> volumeSet,
+                volumeSet -> resourceAttributeUtil.getTypedAttributes(volumeSet, VolumeSetAttributes.class)));
+        LinkedListMultimap<String, Resource> fqdnVolumeSetMultiMap = volumeSetAttributesOptionalMap.entrySet().stream()
+                .filter(entry -> entry.getValue().isPresent()
```

Comment:
Wouldn't you need to filter disks that are already in FAILED state from this pipeline?